plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -183,13 +200,21 @@ static String getParametersString(Map<String, String> parametersMap) {
     return resultBuilder.toString();
   }
 
-  public static void showOptInDialog() {
+  /**
+   * @param parentShell if null, shows the dialog at the workbench level.
+   */
+  public static void showOptInDialog(Shell parentShell) {
     if (!hasUserOptedIn() && !hasUserRegisteredOptInStatus()) {
-      IWorkbenchWindow window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
-      if (window != null) {
-        new OptInDialog(window.getShell()).open();
+      if (parentShell != null) {
```

Comment:
For now, I left out the `Display` logic in the new `findShell()` method. I also catch `IllegalStateException` that `getWorkbench()` might throw. (This exception will be very usual in practice, but I realized it can happen for example when the IDE is being launched.)